<!-- BEGIN KEY -->
public interface java.security.PrivilegedExceptionAction
<!-- END KEY -->
<!-- BEGIN DATA -->

在启用特权的情况下将要执行的计算，会抛出一个或多个经过检查的异常。通过在 <code>PrivilegedExceptionAction</code> 对象上调用 <code>AccessController.doPrivileged</code> 执行该计算。此接口只用于那些抛出经过检查的异常的计算；抛出未经过检查的异常的计算应使用 <code>PrivilegedAction</code>。

@see AccessController
@see AccessController#doPrivileged(PrivilegedExceptionAction)
@see AccessController#doPrivileged(PrivilegedExceptionAction, AccessControlContext)
@see PrivilegedAction

<!-- END DATA -->
<!-- BEGIN KEY -->
public T java.security.PrivilegedExceptionAction.run() throws java.lang.Exception
<!-- END KEY -->
<!-- BEGIN DATA -->

执行计算。此方法将在启动优先级后由 <code>AccessController.doPrivileged</code> 调用。

@return 可以表示计算结果的与类有关的值。每个实现 <code>PrivilegedExceptionAction</code> 的类都应该记录是否此值表示某些内容及其表示的内容。
@throws Exception 出现异常情况。每个实现 <code>PrivilegedExceptionAction</code> 的类都应该记录其 run 方法能够抛出的异常。
@see AccessController#doPrivileged(PrivilegedExceptionAction)
@see AccessController#doPrivileged(PrivilegedExceptionAction,AccessControlContext)

<!-- END DATA -->


